File: solutions_5756407898963968_0/C++/GDS/qual_a.cpp
#include <iostream>
#include <vector>
using namespace std;

vector< int > v1(4), v2(4);

void input() {
	int n1, n2;
	cin >> n1;
	n1--;
	for (int i = 0; i < 16; ++i) {
		int tmp;
		cin >> tmp;
		if (n1 == i/4) v1[i%4] = tmp;
	}
	cin >> n2;
	n2--;
	for (int i = 0; i < 16; ++i) {
		int tmp;
		cin >> tmp;
		if (n2 == i/4) v2[i%4] = tmp;
	}
}

int search() {
	bool alreadyfound = false;
	int result;
	for (int i1 = 0; i1 < 4; ++i1) {
		for (int i2 = 0; i2 < 4; ++i2) {
			if (v1[i1] == v2[i2]) {
				if (alreadyfound) return 0;
				else {
					alreadyfound = true;
					result = v1[i1];
				}
			}
		}
	}
	if (not alreadyfound) return -1;
	return result;
}

int main() {
	int T;
	cin >> T;
	for (int i = 1; i <= T; ++i) {
		input();
		int x = search();
		cout << "Case #" << i << ": ";
		if (x == -1) cout << "Volunteer cheated!" << endl;
		else if (x == 0) cout << "Bad magician!" << endl;
		else cout << x << endl;
	}
}

File: solutions_5756407898963968_0/Python/jameswo/qual_a.py
input = open("A-small-attempt0.in","r")
inputlist = input.readlines()
input.close()
length = len(inputlist)
numberofplays = int(inputlist[0])
output = open("A-small-attempt0.out","w+")
case = 1
casepointer = 1

while (case <= numberofplays):
    firstrowlist2 = []
    secondrowlist2 = []
    
    firstrowstring = inputlist[casepointer + int(inputlist[casepointer])]
    #firstrowstring = inputlist[firstrow]
    firstrowlist = firstrowstring.split(" ")
    for s in firstrowlist:
        firstrowlist2.append(int(s)) #firstrowlist2 is list of elements in firstrow
        
    secondrowstring = inputlist[casepointer + 5 + int(inputlist[casepointer+5])]
    #secondrowstring = inputlist[int(secondrow)]
    secondrowlist = secondrowstring.split(" ")
    for s in secondrowlist:
        secondrowlist2.append(int(s)) #secondrowlist2 is list of elements in secondrow
    matches = []
    for element1 in firstrowlist2:
        for element2 in secondrowlist2:
            if element1 == element2:
                matches.append(element1)
    
    
    if len(matches) == 0:
        output.write("Case #{0}: Volunteer cheated!\n".format(case))
        case += 1
        casepointer +=10
        continue
    
    if len(matches) >= 2:
        output.write("Case #{0}: Bad magician!\n".format(case))
        case += 1
        casepointer +=10
        continue
    
    output.write("Case #{0}: {1}\n".format(case, matches[0]))
               
    case += 1
    
    casepointer += 10

output.close()
